src/store/modules/actions/user.js

```diff
@@ -89,6 +89,7 @@ export async function updatePreferences(
     commit(types.USER_UPDATE_USER_INFO, { creatorPitch });
   }
   if (Object.keys(preferences).length) {
-    await this.$api.$post(api.userPreferences(), preferences);
+    // Wait until the correct token can be obtained and then carry loginStatus = true
+    if (loginStatus) await this.$api.$post(api.userPreferences(), preferences);
```

Comment:
Instead of a new flag, please use the login status in user store
https://github.com/likecoin/liker-land/blob/master/src/store/modules/user.js#L14
to determine if you want to call `updatePreferences` at all or guard this function